### PR TITLE
Edit for redundant information and sections across Data Prepper

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Address benchmark study findings about removing redundant information across Data Prepper docs. Consolidates information into one source doc.  

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:82]: Should we explain the use like we do for equality operators above?

**Comment by vagimeli** [comment]
@dlvenable This PR addresses the benchmark study about redundant documentation across Data Prepper. Please review the changes as soon as possible so that we can close this action item. cc:  @hdhalter 

Tasks completed include consolidating content, rewriting content for comprehensiveness and clarity, and deleting redundant content. The Expression syntax doc has been updated for better organization of information.

File: _data-prepper/index.md
Comment (vagimeli) [inline _data-prepper/index.md:30]:
```suggestion
Each pipeline contains two required components: `source` and `sink`. If a `buffer`, a `processor`, or both are missing from the pipeline, Data Prepper uses the default `bounded_blocking` buffer and a no-op processor. Note that a single instance of Data Prepper can have one or more pipelines. 
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:79]:
```suggestion
The following are some examples of using conditional expressions: 
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:63]:
```suggestion
The following are some examples of using equality operators:
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [reply _data-prepper/pipelines/expression-syntax.md:82]: I don't know if this is necessary. It is fairly standard boolean logic.

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:97]: These (set initializers) don't actually work. We should remove them from the documentation.

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:227]: I think we should keep this format because of the complexity.

And honestly, I'd like to move toward using sub-pages for functions, similar to what we do for processors.

```
Functions
  cidrContains
  contains
  getMetadata
  hasTags
  join
  length
```

File: _data-prepper/pipelines/pipelines.md
Comment (dlvenable) [inline _data-prepper/pipelines/pipelines.md:15]: I think we should stick with "pipeline" as the term here, not "ingest process".

File: _data-prepper/pipelines/pipelines.md
Comment (dlvenable) [inline _data-prepper/pipelines/pipelines.md:89]: These are all covered by common use cases pages right?

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: _data-prepper/pipelines/expression-syntax.md
Comment (Naarcha-AWS) [reply _data-prepper/pipelines/expression-syntax.md:227]: Do we plan on adding additional functions at @dlvenable? I originally had them as section in the functions page so that they were link-able, however, I'm not sure there's enough info on each individual function to justify sub-pages, unless we plan on adding more information to each.

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:15]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/data-prepper-pipeline.png" alt="Data Prepper pipeline">{: .img-fluid}
```

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [reply _data-prepper/pipelines/pipelines.md:89]: @dlvenable Yes, I deleted these from this file and linked the users to the common use cases.

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:100]: Deleted this example because it's redundant to https://github.com/opensearch-project/documentation-website/pull/7127

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:142]: Deleted this example because it's redundant to https://opensearch.org/docs/latest/data-prepper/common-use-cases/trace-analytics/#example-trace-analytics-pipeline

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:204]: Deleted this example because it's redundant to https://github.com/opensearch-project/documentation-website/pull/6248 (waiting for approval to merge PR and publish)

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:230]: Deleted this example because it's redundant to https://opensearch.org/docs/latest/data-prepper/common-use-cases/s3-logs/#pipeline-design

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:271]: Deleted this example because it's redundant to https://opensearch.org/docs/latest/data-prepper/migrating-from-logstash-data-prepper/

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:288]: Deleted this example because it's redundant to https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/configuring-data-prepper/

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:309]: Deleted this example because it's redundant to https://opensearch.org/docs/latest/data-prepper/managing-data-prepper/peer-forwarder/

File: _data-prepper/pipelines/pipelines.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines.md:329]: Deleted this example because it's redundant to https://opensearch.org/docs/latest/data-prepper/pipelines/pipelines/

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [reply _data-prepper/pipelines/expression-syntax.md:227]: Moved to subpages.

File: _data-prepper/pipelines/pipelines-configuration-options.md
Comment (vagimeli) [inline _data-prepper/pipelines/pipelines-configuration-options.md:1]: Deleted this file because it's in pipelines.md

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [reply _data-prepper/pipelines/expression-syntax.md:227]: @Naarcha-AWS , Yes, this is something we plan to expand upon and there may be many more.

@vagimeli , Thank you for making sub-pages!

**Review by dlvenable — APPROVED** [review]

File: _data-prepper/pipelines/pipelines.md
Comment (natebower) [inline _data-prepper/pipelines/pipelines.md:68]:
```suggestion
In the following example pipeline, `application-logs` is a named route with a condition set to `/log_type == "application"`. The route uses [Data Prepper expressions](https://github.com/opensearch-project/data-prepper/tree/main/examples) to define the condition. Data Prepper routes events satisfying this condition to the first OpenSearch sink. By default, Data Prepper routes all events to sinks without a defined route, as shown in the third OpenSearch sink of the given pipeline:
```

File: _data-prepper/pipelines/pipelines.md
Comment (natebower) [inline _data-prepper/pipelines/pipelines.md:95]:
```suggestion
- See [Common uses cases]({{site.url}}{{site.baseurl}}/data-prepper/common-use-cases/common-use-cases/) for example configurations.
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md:49]:
```suggestion
For example, to check if the value of the `status_code` field in an event is within the range of successful HTTP responses (200 to 299), you can use the following expression:
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md:157]:
```suggestion
- Integer: Supports values from -2,147,483,648 to 2,147,483,647.
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md:158]:
```suggestion
- Boolean: Supports `true` or `false`.
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md:159]:
```suggestion
- JSON pointer: See [JSON pointers](#json-pointers) for more information.
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md:160]:
```suggestion
- String: Supports valid Java strings.
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md:161]:
```suggestion
- Null: Supports `null` to check if a JSON pointer exists.
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [reply _data-prepper/pipelines/expression-syntax.md:156]: No.

**Review by hdhalter — COMMENTED** [review]
@dlvenable - I'm trying to close this out for Melissa. Can you please accept my changes and address the couple of comments from Nathan? Thanks!

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md]:
```suggestion
JSON pointers can use an extended character set by wrapping the entire pointer in double quotes `""` and escaping characters using the backslash `\`. Note that the `~` and `/` characters are considered part of the pointer path and do not need to be escaped. 

The following are examples of valid JSON pointers: 

- `~0` to represent the literal character 
- `~` or `~1` to represent the literal character `/`
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md]:
```suggestion
The following is an example of using conditional expressions: 
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (hdhalter) [inline _data-prepper/pipelines/expression-syntax.md]:
```suggestion
| `and`, `or`          | Conditional expressions                                | left-to-right |
```

File: _data-prepper/pipelines/configuration/sources/sources.md
Comment (hdhalter) [reply _data-prepper/pipelines/configuration/sources/sources.md]: I don't think it's not possible, since the Related Articles section is auto-generated.

**Review by hdhalter — COMMENTED** [review]
There were a few more comments that needed to be addressed.

**Comment by natebower** [comment]
@hdhalter Re: "I don't think it's not possible, since the Related Articles section is auto-generated", can we not link to the heading? If not, when referencing the heading, it should be in sentence case: "Related articles".

**Comment by hdhalter** [comment]
> @hdhalter Re: "I don't think it's not possible, since the Related Articles section is auto-generated", can we not link to the heading? If not, when referencing the heading, it should be in sentence case: "Related articles".

Oops, sorry for the double negative! I don't think that section (Next steps) is necessary because it is a small topic and the Related articles section follows it. I'm going to remove it altogether.

File: _data-prepper/pipelines/configuration/buffers/buffers.md
Comment (dlvenable) [reply _data-prepper/pipelines/configuration/buffers/buffers.md:15]: Those two are the only buffers currently supported. It may make sense to link to: https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/buffers/buffers/

File: _data-prepper/pipelines/configuration/sources/sources.md
Comment (dlvenable) [reply _data-prepper/pipelines/configuration/sources/sources.md]: I think we should let the individual sources link to those external pages.

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [reply _data-prepper/pipelines/expression-syntax.md]: A "primary" is defined below, as is an "operator." It would make sense to italicize those.

It may help to see how it is formatted in the original source. You'll see they italicize these terms and that makes it clearer.

https://github.com/opensearch-project/data-prepper/blob/main/docs/expression_syntax.md#definitions

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [reply _data-prepper/pipelines/expression-syntax.md]: A "primary" is defined below, as is an "operator." It would make sense to italicize those.

It may help to see how it is formatted in the original source. You'll see they italicize these terms and that makes it clearer.

https://github.com/opensearch-project/data-prepper/blob/main/docs/expression_syntax.md#definitions

Comment (dlvenable) [reply _data-prepper/pipelines/expression-syntax.md]: I think "immediate" fits here.

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [reply _data-prepper/pipelines/expression-syntax.md]: This is the priority of evaluating the expression tree when parsing.

File: _data-prepper/pipelines/configuration/buffers/buffers.md
Comment (hdhalter) [reply _data-prepper/pipelines/configuration/buffers/buffers.md]: This is the buffer.md page, lol. I'll remove the line for now, until we can point to somewhere that describes how to configure alternative types.

**Comment by hdhalter** [comment]
@natebower - all comments have been addressed and all links are fixed. Do you want to take another pass at it?

**Comment by natebower** [comment]
> @natebower - all comments have been addressed and all links are fixed. Do you want to take another pass at it?

@hdhalter If all comments/suggestions have been addressed, then I won't need to re-review. Thanks.

**Pull request merged**
